VMEncryption/main/handle.py

```diff
@@ -918,16 +918,19 @@ def enable_encryption():
                         passphrase_file = bek_util.get_bek_passphrase_file(encryption_config)
                         crypt_mount_config_util = CryptMountConfigUtil(logger=logger, encryption_environment=encryption_environment, disk_util=disk_util)
                         resource_disk_util = ResourceDiskUtil(logger, disk_util, crypt_mount_config_util, passphrase_file, get_public_settings(), DistroPatcher.distro_info)
-                        rd_encrypted = resource_disk_util.encrypt_resource_disk()
-                        if not rd_encrypted:
-                            hutil.save_seq()
-                            hutil.do_exit(exit_code=CommonVariables.configuration_error,
-                                          operation='EnableEncryption',
-                                          status=CommonVariables.extension_error_status,
-                                          code=str(CommonVariables.configuration_error),
-                                          message='Failed to encrypt resource disk. Please make sure no process is using it.')
+                        if resource_disk_util._resource_disk_exists():
```

Comment:
Resource disk encryption is also called at this location:
https://github.com/Azure/azure-linux-extensions/blob/ade-singlepass-dev/VMEncryption/main/handle.py#L921
Similar check needs to be added there.

Also, I will suggest to move the check to encrypt_format_mount method (https://github.com/Azure/azure-linux-extensions/blob/ade-singlepass-dev/VMEncryption/main/ResourceDiskUtil.py#L282)
The first check here should be if Resource disk exists or not. If it does not exist return True.

Reply:
+1 to Vimish's comments

Reply:
In that case I'd say move the logging lines to the ResourceDiskUtil as well

Reply:
moreover, you might need to add the "resource_disk_exists" check to resource disk util's automount method.

Reply:
Finally moving it all to RDUtil will make writing unit tests a lot easier as that file has a test-file already set up for it

Reply:
> moreover, you might need to add the "resource_disk_exists" check to resource disk util's automount method.

The automount method when invoked without resource disk will return false and won't throw exception. Hence I did not add the check here as the existing code suffices this.

Reply:
> The automount method when invoked without resource disk will return false and won't throw exception. Hence I did not add the check here as the existing code suffices this.

Cool. Could you add a test case for automount to cover that code path?

Reply:
Check for this line in handle.py **rd_encrypted = resource_disk_util.encrypt_resource_disk()**
This will be called when encrypt format all is set while invoking the extension for the first time

Reply:
Sorry...please disregard my comment. I got confused with automount error.

But I still want to either have this check for automount also or move it to encrypt_format_mount.
Even though we don't throw any exception in automount failure we will still go ahead and modify waagent configuration before failing.

Reply:
> Finally moving it all to RDUtil will make writing unit tests a lot easier as that file has a test-file already set up for it

I agree with you on this. Changes have been made for this.

Reply:
> moreover, you might need to add the "resource_disk_exists" check to resource disk util's automount method.

this check has been added at the beginning of encrypt_format_mount() since this function calls the prepare function for resource disk